src/podofo/main/PdfEncoding.cpp

```diff
@@ -612,13 +615,17 @@ void PdfEncoding::writeCIDMapping(PdfObject& cmapObj, const PdfFont& font, const
             range.LastCode.WriteHexTo(temp);
             output.Write(temp);
         }
+
+        output.Write("\nendcodespacerange\n");
     }
     else
     {
+        output.Write("1 begincodespacerange\n");
```

Comment:
This seems wrong as your now writing the "header" of the code space range directly in `AppendCodeSpaceRange(output, temp)`

Reply:
Yes, it's a mistake. I've removed that line of code.